components/service/glean/src/main/java/mozilla/components/service/glean/private/PingType.kt

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.private
+
+import mozilla.components.service.glean.Glean
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * This implements the developer facing API for custom pings.
+ *
+ * Instances of this class type are automatically generated by the parsers at build time.
+ *
+ * The Ping API only exposes the [send] method, which schedules a ping for sending.
+ */
+data class PingType(
+    val name: String,
+    val includeClientId: Boolean
+) {
+    init {
+        if (pingRegistry.containsKey(name)) {
+            logger.error("Duplicate ping named $name")
+        }
+        pingRegistry[name] = this
+    }
+
+    companion object {
+        private val logger = Logger("glean/PingType")
+        internal val pingRegistry: MutableMap<String, PingType> = mutableMapOf()
+    }
+
+    /**
+     * Send the ping.
+     *
+     * While the collection of metrics into pings happens synchronously, the
```

Comment:
nit: mh, I'd rephrase this a bit. After the last change to `sendPings`, collection happens asynchronously as well. What about.. "The ping is assembled as soon as possible, but upload is not guaranteed to happen immediately as it depends on the upload policies."